proposals/0357-alpenglow_validator_admission_ticket.md

```diff
@@ -0,0 +1,191 @@
+
+Before Alpenglow, we have the following accounts related to voting:
+
+- Vote account for saving all the vote states and receiving commission
+
+- Identity account for receiving block rewards and priority fees, the vote
+transaction fees are currently paid out of this account
+
+After Alpenglow, we will still have roughly the same accounts serving the same
+purpose:
+
+- Vote account: This account must contain the correct BLS public key
+corresponding to vote authority keypair. It continues to keep all the vote
+credits and all validator identity/authority/commission information updates
+happen here, but the `votes` list in vote state will be empty. This will be
+the account where the 1.6 SOL VAT is deducted from.
+
+- Identity account: It continues receiving block rewards and priority fees.
+But we may change that in a separate SIMD.
+
+## Detailed Design
+
+### What is considered a valid vote account
+
+A valid vote account in an Alpenglow epoch must contain:
+
+- a BLS public key
+
+- at least 1.6 SOL VAT fee plus the necessary storage rent amount for a new
+epoch in the vote account
+
+When the staked validators for a new epoch are calculated, all validators must
+perform the following operations:
+
+- When there are more than 2,000 valid validators, sort all valid vote accounts
+by descending order of stake. If some validator with stake S is in position
+2001, then we remove all validators with stake S and less. If there are fewer
+than 2,000 valid validators, pick all of them.
+
+- Deduct 1.6 SOL VAT fee from each picked vote account once
+
+- Mark the fee burned and write the result into the bank
+
+### How to implement the checks
+
+1. When a new bank crosses an epoch boundary (bank.epoch() >
+parent_bank.epoch()), calculate the participating staked validators for the
+next epoch (bank.epoch() + 1). This is the same as now.
+
+2. Perform stake activation and deactivation so that the intended stake values
+are used for the new epoch. This operation occurs before any transactions are
+processed in the new epoch’s bank.
+
+3. The calculation iterates all vote accounts and filters those that meet
+the following criteria:
+
+  - The account has a balance of at least 1.6 SOL
+
+  - The account has a BLS public key
+
+4. If the number of filtered accounts exceeds 2,000, sort them according to the
+following rules and select the top 2,000. Otherwise, return the entire list:
+
+  - Sort in descending order of stake (largest to smallest)
+
+  - If multiple validators have exactly the same amount of stake and including
+  all of them would exceed the 2,000 limit, then exclude all of them
+
+5. Subtract 1.6 SOL from the vote account for each validator in the accepted
+list from the previous step
+
+6. Record the VAT fee subtraction in the bank, which reduces the bank’s
+capitalization. The recording of fee subtraction occurs when the bank is
+frozen even though fee reduction happens before any transaction in the
+bank is executed.
```

Comment:
these steps aren't particularly precise. we should probably just say to move the lamports directly into the incinerator account?